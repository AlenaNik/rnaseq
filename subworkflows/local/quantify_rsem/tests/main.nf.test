nextflow_workflow {

    name "Test Workflow QUANTIFY_RSEM"
    script "../main.nf"
    workflow "QUANTIFY_RSEM"
    config "./nextflow.config"

    tag 'RSEM_CALCULATEEXPRESSION'
    tag 'RSEM_MERGE_COUNTS'
    tag 'BAM_SORT_STATS_SAMTOOLS'

    test("homo_sapiens") {

        setup {
            run("RSEM_PREPAREREFERENCE") {
                script "../../../../modules/nf-core/rsem/preparereference/main.nf"
                process {
                    """
                    input[0] = Channel.of(file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true))
                    input[1] = Channel.of(file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true))
                    """
                }
            }
        }

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test', strandedness: 'forward' ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = RSEM_PREPAREREFERENCE.out.index
                input[2] = Channel.of(file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true))
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }

    }

}
