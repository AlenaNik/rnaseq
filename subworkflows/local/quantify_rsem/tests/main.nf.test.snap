{
    "sarscov2": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "10": [
                    
                ],
                "11": [
                    
                ],
                "12": [
                    
                ],
                "13": [
                    
                ],
                "14": [
                    
                ],
                "15": [
                    
                ],
                "16": [
                    
                ],
                "17": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    
                ],
                "bai": [
                    
                ],
                "bam": [
                    
                ],
                "bam_genome": [
                    
                ],
                "bam_star": [
                    
                ],
                "bam_transcript": [
                    
                ],
                "counts_gene": [
                    
                ],
                "counts_transcript": [
                    
                ],
                "csi": [
                    
                ],
                "flagstat": [
                    
                ],
                "idxstats": [
                    
                ],
                "logs": [
                    
                ],
                "merged_counts_gene": [
                    
                ],
                "merged_counts_transcript": [
                    
                ],
                "merged_tpm_gene": [
                    
                ],
                "merged_tpm_transcript": [
                    
                ],
                "stat": [
                    
                ],
                "stats": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-07T14:32:58.87423"
    }
}